helpers/lbryApi.js

```diff
@@ -37,7 +37,35 @@ function orderTopClaims(claimsListArray){
 	return claimsListArray;
 }
 
+function getClaimWithUri(uri, resolve, reject){
+	console.log(">> making get request to lbry daemon")
+	axios.post('http://localhost:5279/lbryapi', {
+			method: "get",
+			params: { uri: uri }
+		}
+	).then(function (getUriResponse) {
+		console.log(">> 'get claim' success...");
+		//check to make sure the daemon didn't just time out
+		if (getUriResponse.data.result.error === "Timeout"){
+			reject("get request to lbry daemon timed out");
+		}
+		console.log(">> response data:", getUriResponse.data);
+		console.log(">> dl path =", getUriResponse.data.result.download_path)
+		// resolve the promise with the download path for the claim we got
+		/* 
+			note: put in a check to make sure we do not resolve until the download is actually complete 
+		*/
+		resolve(getUriResponse.data.result.download_path);
```

Comment:
If you find yourself accessing the same field repeatedly, like getUriResponse.data, it can be good to just dereference it.

Additionally, assuming you're writing ES6 (which is probably a good idea), you can just write:

`.then(({data}) => {
  if (data.result.error === "Timeout")
  //etc.
})`